File: EscapeFromTrinityEngineStats/Models/RewardRecord.cs
using EscapeFromTrinityEngineStats.Models.InputDto;
using EscapeFromTrinityEngineStats.Models.Relationships;
using System.Text.Json.Serialization;

namespace EscapeFromTrinityEngineStats.Models
{
    public class RewardRecord
    {
        public int Id { get; set; }
        public int VersionId { get; set; }

        [JsonIgnore]
        public virtual GameVersion Version { get; set; }

        public int GoldGained { get; set; }
        public virtual List<CardChoiceRecord> CardChoiceRecords { get; set; } = [];
        public virtual List<PassiveRecord> PassiveRecords { get; set; } = [];
        public virtual List<RewardUpgradedCards> UpgradedCards { get; set; } = [];
        public virtual List<RewardRemovedCards> RemovedCards { get; set; } = [];
        public virtual List<RewardGivenTradeCards> GivenTradeCards { get; set; } = [];
        public virtual List<RewardRecievedTradeCards> RecievedTradeCards { get; set; } = [];
        public virtual List<RewardJunkCards> JunkRewards { get; set; } = [];
    }
}

File: EFTEDashboard/src/app/interfaces/reward-record.ts
import { CardChoiceRecord } from './card-choice-record';
import { GameVersion } from './game-version';
import { PassiveRecord } from './passive-record';
import { RewardGivenTradeCards } from './relationships/reward-given-trade-cards';
import { RewardJunkCards } from './relationships/reward-junk-cards';
import { RewardRecievedTradeCards } from './relationships/reward-recieved-trade-cards';
import { RewardRemovedCards } from './relationships/reward-removed-cards';
import { RewardUpgradedCards } from './relationships/reward-upgraded-cards';

export interface RewardRecord {
	id?: number;
	versionId?: number;
	version?: GameVersion;
	passiveRecords?: PassiveRecord[];
	cardChoiceRecords?: CardChoiceRecord[];
	upgradedCards?: RewardUpgradedCards[];
	removedCards?: RewardRemovedCards[];
	givenTradeCards?: RewardGivenTradeCards[];
	recievedTradeCards?: RewardRecievedTradeCards[];
	junkRewards?: RewardJunkCards[];
}
